File: src/JavaScript/frequency/1.js

function frequency(sentence) {
  let count = {}
  let maxCount = 0
  let maxChar = ''

  let formatedSentence = sentence.replace(/\s+/g, '').toLowerCase()

  for (let char of formatedSentence) {
    count[char] = (count[char] || 0) + 1
    if (count[char] > maxCount) {
      maxCount = count[char]
      maxChar = char
    }
  }
  return [maxChar, maxCount]
}

let result = 'This is a common interview question'
console.log(frequency(result))









const sample = "This is a common interview question";

File: src/Python/ frequency/1.py
# Crea un algoritmo para encontrar la frecuencia de caracteres en una cadena de texto
# Parámetro: sentence, cadena de texto
# Retorno: el caracter con mayor frecuencia
# Input example: ('This is a common interview question')
# output example: ('i', 5)


def frequency(sentence):
    formated_sentence = sentence.replace(' ', '').lower()

    char_frequency = {char: formated_sentence.count(
        char) for char in formated_sentence}

    char_frequency_sorted = sorted(
        char_frequency.items(), key=lambda kv: kv[1], reverse=True)

    return char_frequency_sorted[0]


sample = "This is a common interview question"
print(frequency(sample))
